zerver/tests/test_upload.py

```diff
@@ -212,7 +215,27 @@ def test_preserve_provided_content_type(self) -> None:
         url = response_dict["url"]
         result = self.client_get(url)
         self.assertEqual(result.status_code, 200)
-        self.assertEqual(result["Content-Type"], "image/png")
+        self.assertEqual(result["Content-Type"], 'text/plain; charset="utf-8"')
+        consume_response(result)
+
+    def test_content_type_charset_specified(self) -> None:
+        with tempfile.NamedTemporaryFile() as uploaded_file:
+            uploaded_file.write("नाम में क्या रक्खा हे".encode())
```

Comment:
Should this be `encode('big5')` or similar?

Reply:
No.  The goal is to test that we trust the encoding that the client specifies, and not always use the auto-detection algorithm.

Reply:
I'm pretty sure that those characters also don't exist in `big5`, too. :)